selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/git/GitPullConflictTest.java

```diff
@@ -0,0 +1,196 @@
+          + "src/main/java/org/eclipse/qa/examples/AppController.java\n"
+          + "README.md\n"
+          + "Automatic merge failed; fix conflicts and then commit the result.";
+
+  private static final String CHANGE_STRING_1 =
+      String.format("//first_change_%s", System.currentTimeMillis());
+
+  private static final String HEAD_CONF_PREFIX_CONF_MESS =
+      String.format("<<<<<<< HEAD\n//second_change\n=======\n%s\n>>>>>>>", CHANGE_STRING_1);
+
+  private GitHub gitHub;
+  private GHRepository gitHubRepository;
+
+  @Inject private TestWorkspace ws;
+  @Inject private Ide ide;
+  @Inject private TestUser productUser;
+
+  @Inject
+  @Named("github.username")
+  private String gitHubUsername;
+
+  @Inject
+  @Named("github.password")
+  private String gitHubPassword;
+
+  @Inject private TestProjectServiceClient testProjectServiceClient;
+  @Inject private ProjectExplorer projectExplorer;
+  @Inject private Menu menu;
+  @Inject private Git git;
+  @Inject private Events events;
+  @Inject private Loader loader;
+  @Inject private CodenvyEditor editor;
+  @Inject private Consoles consoles;
+  @Inject private TestSshServiceClient testSshServiceClient;
+  @Inject private TestUserPreferencesServiceClient testUserPreferencesServiceClient;
+  @Inject private TestGitHubServiceClient gitHubClientService;
+
+  @BeforeClass
+  public void prepare() throws Exception {
+    gitHub = GitHub.connectUsingPassword(gitHubUsername, gitHubPassword);
+    gitHubRepository = gitHub.createRepository(REPO_NAME).create();
+    String commitMess = String.format("add-new-content %s ", System.currentTimeMillis());
+    testUserPreferencesServiceClient.addGitCommitter(gitHubUsername, productUser.getEmail());
+    Path entryPath =
+        Paths.get(getClass().getResource("/projects/default-spring-project").getPath());
+    gitHubClientService.addContentToRepository(entryPath, commitMess, gitHubRepository);
+    ide.open(ws);
+  }
+
+  @AfterClass
+  public void deleteRepo() throws IOException {
+    gitHubRepository.delete();
+  }
+
+  @Test
+  public void pullConflictsTest() throws Exception {
+    // preconditions and import the test repo
+    String javaFileChange = "AppController";
+    String textFileChange = "README.md";
+    String changeContent2 = "//second_change";
+
+    projectExplorer.waitProjectExplorer();
+    String repoUrl = String.format("https://github.com/%s/%s.git", gitHubUsername, REPO_NAME);
+    git.importJavaApp(repoUrl, PROJECT_NAME, MAVEN);
+
+    // change files in the test repo on GitHub
+    changeContentOnGithubSide(
+        String.format("%s/%s.java", PATH_TO_JAVA_FILE, javaFileChange), CHANGE_STRING_1);
+    changeContentOnGithubSide(textFileChange, CHANGE_STRING_1);
+
+    // change the same files in the editor
+    changeJavaFileForTest(javaFileChange, changeContent2);
+    changeTextFileForTest(textFileChange, changeContent2);
+
+    // make pull and get the first conflict
```

Comment:
```// make pull and get the first conflict ->  make pull and check the expected conflict message``` or something like that

Reply:
Actually, we aren't checking expected conflict message. We are checking if conflict message we get is the same as expected.

Reply:
```make pull and get the first conflict`` looks strange as for me